File: asyncapi-core/src/test/kotlin/com/asyncapi/bindings/websockets/v0/_1_0/channel/WebSocketsChannelBindingTest.java
package com.asyncapi.bindings.websockets.v0._1_0.channel;

import com.asyncapi.bindings.websockets.v0._1_0.WebSocketsBindingProvider;
import com.asyncapi.v3.SerDeTest;
import org.jetbrains.annotations.NotNull;
import org.junit.jupiter.api.Nested;

@Nested
public abstract class WebSocketsChannelBindingTest extends SerDeTest<WebSocketsChannelBinding> {

    @NotNull
    @Override
    protected Class<WebSocketsChannelBinding> objectClass() {
        return WebSocketsChannelBinding.class;
    }

    @NotNull
    @Override
    protected String baseObjectJson() {
        return "/bindings/websockets/0.1.0/channel/webSocketsChannelBinding.json";
    }

    @NotNull
    @Override
    protected String extendedObjectJson() {
        return "/bindings/websockets/0.1.0/channel/webSocketsChannelBinding - extended.json";
    }

    @NotNull
    @Override
    protected String wronglyExtendedObjectJson() {
        return "/bindings/websockets/0.1.0/channel/webSocketsChannelBinding - wrongly extended.json";
    }

    @NotNull
    @Override
    public WebSocketsChannelBinding build() {
        return WebSocketsBindingProvider.channel();
    }

}

File: asyncapi-core/src/test/kotlin/com/asyncapi/bindings/websockets/v0/_1_0/channel/WebSocketsChannelBindingTest.kt
package com.asyncapi.bindings.websockets.v0._1_0.channel

import com.asyncapi.v3.SerDeTest
import com.asyncapi.v3.schema.AsyncAPISchema
import com.asyncapi.v3.schema.Type


/**
 * @version 3.0.0
 * @author Pavel Bodiachevskii
 */
class WebSocketsChannelBindingTest: SerDeTest<WebSocketsChannelBinding>() {

    override fun objectClass() = WebSocketsChannelBinding::class.java

    override fun baseObjectJson() = "/bindings/websockets/channel/webSocketsChannelBinding.json"

    override fun extendedObjectJson() = "/bindings/websockets/channel/webSocketsChannelBinding - extended.json"

    override fun wronglyExtendedObjectJson() = "/bindings/websockets/channel/webSocketsChannelBinding - wrongly extended.json"

    override fun build(): WebSocketsChannelBinding {
        return WebSocketsChannelBinding.builder()
                .method(WebSocketsChannelMethod.GET)
                .query(AsyncAPISchema.builder()
                        .type(Type.OBJECT)
                        .properties(mapOf(
                                Pair(
                                        "ref",
                                        AsyncAPISchema.builder()
                                                .type(Type.STRING)
                                                .description("Referral.")
                                                .build()
                                )
                        ))
                        .build()
                )
                .headers(AsyncAPISchema.builder()
                        .type(Type.OBJECT)
                        .properties(mapOf(
                                Pair(
                                        "Authentication",
                                        AsyncAPISchema.builder()
                                                .type(Type.STRING)
                                                .description("Authentication token")
                                                .build()
                                )
                        ))
                        .build()
                )
                .build()
    }

}